File: Utility/Logger.cs
using System;
using System.Drawing;
using Pastel;

namespace UOToolBox.Utility
{
    public class Logger
    {
        // Implement dynamic change by config.
        public static int ModuleFilter = (int)(Module.LocalFileSystem | Module.SignalR);
        
        [Flags]
        public enum Module
        {
            PipeClient = 1 << 1, 
            Parser = 1 << 2, 
            LocalFileSystem = 1 << 3, 
            SignalR = 1 << 4,
        }
        
        public static void Log(Module module, string message)
        {
            var value = ModuleFilter & (int) module;
            if (value == (int) module)
            {
                Console.WriteLine($"[{$"UTB-Server\t| {Enum.GetName(module)}".Pastel(Color.DarkGreen)}] {message}");
            }
        }
    }
}

File: ClientApp/src/utility/logger.ts
export enum LoggerModule {
  SignalR = 1 << 1
}

export class Logger
{
  // Implement dynamic change by config.
  public static moduleFilter = LoggerModule.SignalR;

  public static Log(module: LoggerModule , message: string)
  {
    const value = Logger.moduleFilter & module;
    if (value == module)
    {
      console.log(`[UTB-Client\t| ${LoggerModule[module]}] ${message}`);
    }
  }
}